packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -208,32 +210,37 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
         isUsingMockResponse(appContext?.application?.id);
       setIsLoading(true);
       try {
-        const res = await DataFetcher.fetch(
-          api,
-          {
-            ...evaluatedInputs,
-            ...context,
-            ensemble: {
-              env: appContext?.env,
-              secrets: appContext?.secrets,
-            },
-          },
-          {
-            mockResponse: mockResponse(
-              mockResponses[api.name],
-              useMockResponse,
+        const response = await queryClient.fetchQuery({
+          queryKey: [api.name],
```

Comment:
This needs to be a hash of screen/name/inputs. Otherwise the invoking the same API again on a different screen or different inputs will return the wrong result.

Add a test for this.